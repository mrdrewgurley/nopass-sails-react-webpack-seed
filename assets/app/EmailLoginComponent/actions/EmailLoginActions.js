import { createAction } from 'redux-actions'
import * as types from '../constants/EmailLoginActionTypes'

export const attemptHandshake = createAction(types.ATTEMPT_HANDSHAKE)
export const handshakeFailure = createAction(types.HANDSHAKE_FAILURE)
export const handshakeSuccess = createAction(types.HANDSHAKE_SUCCESS)
