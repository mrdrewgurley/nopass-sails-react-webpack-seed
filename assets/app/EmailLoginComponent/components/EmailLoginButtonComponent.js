import axios from 'axios'
import classNames from 'classnames'
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class EmailLoginButtonComponent extends PureComponent {

  requestAccess = () => {
    var that = this
    axios
      .put('/auth/handshake', {
        email: document.getElementById('form-input-email').value
      })
      .then(res => {
        if (res.status == 204) {
          that.props.Actions.handshakeSuccess()
          window.uikit.notification({
            message: "<span uk-icon='icon: info'></span><span class='text'>Success, please check your email inbox</span>",
            status: 'success'
          })
          document.getElementById('EmailLoginInput').remove();
        }
      })
      .catch(function(err) {
        if (err.response.status == 400) {
          that.props.Actions.handshakeFailure()
          err.response.data.forEach(message => {
            window.uikit.notification({
              message: "<span uk-icon='icon: warning'></span><span class='text'>" + message + "</span>",
              status: 'danger'
            })
          })
        }
      })
  }

  handleClick = (e) => {
    e.preventDefault()
    this.props.Actions.attemptHandshake()
    this.requestAccess()
  }

  render() {
    return (
      <div className="uk-button-group">
        <button
          className={classNames('uk-button', this.props.EmailLoginButtonReducer.class)}
          onClick={(e) => this.handleClick(e)}
          disabled={!this.props.EmailLoginButtonReducer.state}
        >
          {this.props.EmailLoginButtonReducer.label}
        </button>
      </div>
    )
  }

}

EmailLoginButtonComponent.propTypes = {

  Actions: PropTypes.object.isRequired,
  EmailLoginButtonReducer: PropTypes.object.isRequired

}
