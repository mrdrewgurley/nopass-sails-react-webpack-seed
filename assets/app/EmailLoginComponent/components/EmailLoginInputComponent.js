import classNames from 'classnames'
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class EmailLoginInputComponent extends PureComponent {
  render() {
    return (
      <div className="uk-inline" id="EmailLoginInput">
        <span className="uk-form-icon" uk-icon="icon: user"></span>
        <input
          id="form-input-email"
          className={classNames('uk-input', this.props.EmailLoginInputReducer.class)}
          disabled={!this.props.EmailLoginInputReducer.state}
          required
        />
        <label htmlFor="form-input-email">Email Address</label>
      </div>
    )
  }
}

EmailLoginInputComponent.propTypes = {
  Actions: PropTypes.object.isRequired,
  EmailLoginInputReducer: PropTypes.object.isRequired
}
