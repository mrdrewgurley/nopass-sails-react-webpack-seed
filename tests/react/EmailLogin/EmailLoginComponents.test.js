import React from 'react'
import renderer from 'react-test-renderer'

import EmailLoginButtonComponent from '../../../assets/app/EmailLoginComponent/components/EmailLoginButtonComponent'
import EmailLoginInputComponent from '../../../assets/app/EmailLoginComponent/components/EmailLoginInputComponent'
import * as Actions from '../../../assets/app/EmailLoginComponent/actions/EmailLoginActions'

/**
 * React.EmailLoginComponents
 * @desc Tests for the Email Login Components
 */
describe("Email Login Components", () => {
  describe("Email Login Button Component", () => {
    it("renders with proper defaults", () => {
      const EmailLoginButton = renderer
        .create(<EmailLoginButtonComponent
          Actions={Actions}
          EmailLoginButtonReducer={{}} />)
        .toJSON()

      expect(EmailLoginButton).toMatchSnapshot()
    })
  })

  describe("Email Login Button Component", () => {
    it("renders with proper defaults", () => {
      const EmailLoginInput = renderer
        .create(<EmailLoginInputComponent
          Actions={Actions}
          EmailLoginInputReducer={{}} />)
        .toJSON()

      expect(EmailLoginInput).toMatchSnapshot()
    })
  })
})
