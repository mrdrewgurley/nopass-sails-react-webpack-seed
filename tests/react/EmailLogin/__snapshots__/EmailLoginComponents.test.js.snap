// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Login Components Email Login Button Component renders with proper defaults 1`] = `
<div
  className="uk-button-group"
>
  <button
    className="uk-button"
    disabled={true}
    onClick={[Function]}
  />
</div>
`;

exports[`Email Login Components Email Login Button Component renders with proper defaults 2`] = `
<div
  className="uk-inline"
  id="EmailLoginInput"
>
  <span
    className="uk-form-icon"
    uk-icon="icon: user"
  />
  <input
    className="uk-input"
    disabled={true}
    id="form-input-email"
    required={true}
  />
  <label
    htmlFor="form-input-email"
  >
    Email Address
  </label>
</div>
`;
